packages/config/src/api/site_info.js

```diff
@@ -15,26 +17,35 @@ export const getSiteInfo = async function ({
   siteId,
   mode,
   siteFeatureFlagPrefix,
+  featureFlags = {},
   testOpts: { env: testEnv = true } = {},
 }) {
   if (api === undefined || mode === 'buildbot' || !testEnv) {
     const siteInfo = siteId === undefined ? {} : { id: siteId }
     return { siteInfo, accounts: [], addons: [] }
   }
+  const fetchIntegrations = featureFlags.buildbot_fetch_integrations
+
+  const promises = [getSite(api, siteId, siteFeatureFlagPrefix), getAccounts(api), getAddons(api, siteId)]
+
+  if (fetchIntegrations) {
+    promises.push(getIntegrations(api, 'site', siteId))
+  }
 
-  const [siteInfo, accounts, addons] = await Promise.all([
-    getSite(api, siteId, siteFeatureFlagPrefix),
-    getAccounts(api),
-    getAddons(api, siteId),
-  ])
+  const [siteInfo, accounts, addons, ...rest] = await Promise.all(promises)
```

Comment:
Is there a more descriptive name than `rest` that we can use here? Can we destructure the 4th element into a variable like `integrations`, and then check if it's `undefined`?

Reply:
yeah I think I just forgot how well the array destructuring works, as that is a far nicer approach